artemis-protocols/artemis-amqp-protocol/src/main/java/org/apache/activemq/artemis/protocol/amqp/broker/AMQPLargeMessage.java

```diff
@@ -362,6 +348,62 @@ public void addBytes(ReadableBuffer data) throws Exception {
       }
    }
 
+   protected void parseLargeMessage(ActiveMQBuffer data, boolean initialHeader) {
+      MessageDataScanningStatus status = getDataScanningStatus();
+      if (status == MessageDataScanningStatus.NOT_SCANNED) {
+         ByteBuf buffer = data.byteBuf().duplicate();
+         if (parsingBuffer == null) {
+            parsingBuffer = new CompositeReadableBuffer();
+         }
+
+         byte[] parsingData = new byte[buffer.readableBytes()];
+         buffer.readBytes(parsingData);
+         parsingBuffer.append(parsingData);
+         if (!initialHeader) {
+            genericPargeLargeMessage();
+         }
+      }
+   }
+
+   protected void parseLargeMessage(byte[] data, boolean initialHeader) {
+      MessageDataScanningStatus status = getDataScanningStatus();
+      if (status == MessageDataScanningStatus.NOT_SCANNED) {
+         byte[] copy = new byte[data.length];
+         System.arraycopy(data, 0, copy, 0, data.length);
+         if (parsingBuffer == null) {
+            parsingBuffer = new CompositeReadableBuffer();
+         }
+         parsingBuffer.append(copy);
+         if (!initialHeader) {
+            genericPargeLargeMessage();
+         }
+      }
+   }
+
+   private void genericPargeLargeMessage() {
```

Comment:
Thinking there might be a spelling error here

Reply:
yeah.. thanks

Reply:
fixed